MaplePacketLib2/Tools/PacketWriter.cs

```diff
@@ -80,6 +80,11 @@ public PacketWriter WriteMode(Enum value)
             return Write(Convert.ToByte(value));
         }
 
+        public PacketWriter WriteEnum(Enum value)
+        {
+            return Write(Convert.ToInt32(value));
```

Comment:
This will convert all enums to ints, we need to check what type the enum is and convert to that type.

Reply:
That or just do `Write(value);`, it accepts basically anything and it should automatically write the size of the Enum, but if not, I agree, this is a useless method if we need to force all enums to be one specific size.